lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -1203,18 +1220,24 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
 
     final Size chartViewSize = getChartUsableDrawSize(viewSize);
 
-    /// Find the nearest spot (on X axis)
-    for (int i = 0; i < barData.spots.length; i++) {
-      final spot = barData.spots[i];
-      if (spot.isNotNull()) {
-        if ((touchedPoint.dx - getPixelX(spot.x, chartViewSize)).abs() <=
-            data.lineTouchData.touchSpotThreshold) {
-          return LineBarSpot(barData, barDataPosition, spot);
-        }
-      }
-    }
+    final closestSpotByX = barData.spots.reduce((lhs, rhs) {
```

Comment:
Please don't use names such as `lhs` `rhs` these are not descriptive.
Let them be too long, it is way better than `lhs` and `rhs`.